### ISSUE TITLE
CLAS: detect WITH_UNIT_TESTS from the presence of the corresponding file

### ISSUE THREAD

**Issue by filak-sap** [larshp/abapGit#1530]
In our systems, `zcl_abapgit_object_clas` uses `zcl_abapgit_oo_class_new` as a controller for dealing with class sources. The controller creates local defintions, implementations, macros and testclassed via own calls to "insert report". The function expects that the corresponding reports were created before:
https://github.com/larshp/abapGit/blob/master/src/objects/zcl_abapgit_oo_class_new.clas.abap#L285

Unfortunately, the `CCAU` - testclasses are not created by the function `SEO_CLASS_CREATE_COMPLETE` by default and, hence, pulls always crash in our systems.

I can either use the standard function `SEO_CLASS_GENERATE_LOCALS` or I at least must mimic its functionality - add a dummy empty report.

**Comment by jfilak** [comment]
> "pulls always crash in our systems" all pulls that contain CLAS objects crash?

I apologize for being so vague. Let me rephrase the description:

It is not possible to pull a new class with local implementations and test classes - please, note that there is not local definition. Whenever I try to create such a class in pull, I get the error message:
`Fatal error. Include CCAU should have been created previously!`
The error class is generated in the method `update_report` of the class `zcl_abapgit_oo_class_new`.

> What SAP version are you on?

I can provide you with some details from the dialog "System: Kernel information".

```
Kernel Release: 774
SAP Versions: 753, 754, 773, 774
```

**Comment by christianguenter2** [comment]
Tried to reproduce this in my system (751 SP02) but no luck. Can you please provide a public repository with a test case?

**Comment by filak-sap** [comment]
Unfortunately, no longer reproducible with Z classes. Let me try to reproduce it with standard classes.

**Comment by filak-sap** [comment]
Fortunately, the problem was originally discovered with Z classes - I spent 3 hours debugging and testing. Seems to me that someone somehow fixed the problem. If only I had my development environment under control.

BTW: here is the example repo: https://github.com/filak-sap/example_clas_pull (check out branches).

**Comment by filak-sap** [comment]
I have just managed to reproduced the issue. I did not committed the change:
```diff
+ <WITH_UNIT_TESTS>X</WITH_UNIT_TESTS>
```
I thought the parameter is redundant, since we can get the information by checking existence of the *.testclasses.abap file.

**Comment by gregorwolf** [comment]
Adding checks that the tables passed to update_report are not initial in:
ZCL_ABAPGIT_OO_CLASS_NEW->ZIF_ABAPGIT_OO_OBJECT_FNC~GENERATE_LOCALS
fixed the issue for me. Seems @christianguenter2 or @christianguenter3 will investigate further.

**Comment by mbtools** [comment]
No complaints for 1.5 years. Let's close?

**Comment by larshp** [comment]
yea, also think this is something that should be discovered by the linter, closing
